### PR TITLE
1611177: Document that uploadEnabled=false clears metrics

### PR DESCRIPTION
**Pull request by mdboom**

### CODE REVIEW

File: docs/user/general-api.md [hunk 1]
```diff
@@ -50,12 +50,13 @@ class SampleApplication : Application() {
         Glean.registerPings(Pings)
 
         // Initialize the Glean library.
-        Glean.initialize(applicationContext, uploadEnabled = true)
+        Glean.initialize(applicationContext, uploadEnabled = settings().isTelemetryEnabled)
```
Comment (Dexterp37) [inline docs/user/general-api.md]: I'd be tempted to leave this line as it was or explain that `settings().isTelemetryEnabled` is something which is not part of the SDK but whatever contains the user preference/intent for having telemetry enabled.

File: docs/user/general-api.md [hunk 2]
```diff
@@ -50,12 +50,13 @@ class SampleApplication : Application() {
...
     }
 }
 ```
 
-Once initialized, if collection is enabled, the Glean SDK will automatically start collecting [baseline metrics](pings/metrics.md) and sending its [pings](pings/index.md).
+Once initialized, if `uploadEnabled` is true, the Glean SDK will automatically start collecting [baseline metrics](pings/metrics.md) and sending its [pings](pings/index.md).
```
Comment (Dexterp37) [inline docs/user/general-api.md]:
```suggestion
Once initialized, if `uploadEnabled` is true, the Glean SDK will automatically start collecting [baseline metrics](pings/metrics.md) and sending its [pings](pings/index.md), according to their respective schedule.  
```

File: docs/user/general-api.md (hunk 1 above)
Comment (mdboom) [reply docs/user/general-api.md]: I think I prefer to add a comment about it -- the purpose of this change is to make it clear that this value should come from a user preference, and not be a constant (and also to match the Swift example below, but that's secondary)...

Comment (Dexterp37) [reply docs/user/general-api.md]: Comment sounds good (maybe also in the swift example?)

**Pull request merged**